ios/BT/API/APIClient.swift

```diff
@@ -81,14 +108,35 @@ private extension APIClient {
     static let errorMessage = "error_description"
   }
 
+  func downloadRequest<T: APIRequest>(for request: T, fileName: String) -> DownloadRequest {
```

Comment:
Not sure if this is a swift thing, but should there be 2 `downloadRequest` functions?

Reply:
good question - yes, it's idiomatic (depending on who you ask) to have 2 functions with identical names but different signatures. The compiler (autocomplete) helps you out during development to ensure that you're calling the intended one

Reply:
pretty neat! elixir has a similar feature and its awesome!